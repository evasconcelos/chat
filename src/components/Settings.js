import React from 'react';
import styled from 'styled-components';
import l from 'utils/localization';

export default ({ settings, setSettingsItem, resetSettings }) => (
  <Container>
    <h1>{l(`Settings`)}</h1>
    <SettingsItem>
      <label>
        <SettingName>{l(`User name`)}</SettingName>
        <input
          type="text"
          name="userName"
          value={settings.userName}
          maxLength="20"
          onChange={e => {
            setSettingsItem('userName', e.target.value);
          }}
        />
      </label>
    </SettingsItem>

    <SettingsItem>
      <SettingName>{l(`Interface Color`)}</SettingName>
      <label>
        <input
          type="radio"
          name="interfaceColor"
          value="light"
          checked={settings.lightTheme === true}
          onChange={() => setSettingsItem('lightTheme', true)}
        />
        {l(`Light`)}
      </label>
      <label>
        <input
          type="radio"
          name="interfaceColor"
          value="dark"
          checked={settings.lightTheme === false}
          onChange={() => setSettingsItem('lightTheme', false)}
        />
        {l(`Dark`)}
      </label>
    </SettingsItem>

    <SettingsItem>
      <SettingName>{l(`Clock Display`)}</SettingName>
      <label>
        <input
          type="radio"
          name="clockDisplay12h"
          value="12"
          checked={settings.clockDisplay12h === true}
          onChange={() => setSettingsItem('clockDisplay12h', true)}
        />
        {l(`12 Hours`)}
      </label>
      <label>
        <input
          type="radio"
          name="clockDisplay12h"
          value="24"
          checked={settings.clockDisplay12h === false}
          onChange={() => setSettingsItem('clockDisplay12h', false)}
        />
        {l(`24 Hours`)}
      </label>
    </SettingsItem>

    <SettingsItem>
      <SettingName>{l(`Send messages on CTRL+ENTER`)}</SettingName>
      <label>
        <input
          type="radio"
          name="ctrlEnterSendMsg"
          value="yes"
          checked={settings.ctrlEnterSendMsg === true}
          onChange={() => setSettingsItem('ctrlEnterSendMsg', true)}
        />
        {l(`On`)}
      </label>
      <label>
        <input
          type="radio"
          name="ctrlEnterSendMsg"
          value="no"
          checked={settings.ctrlEnterSendMsg === false}
          onChange={() => setSettingsItem('ctrlEnterSendMsg', false)}
        />
        {l(`Off`)}
      </label>
    </SettingsItem>

    <SettingsItem>
      <label>
        <SettingName>{l(`Language`)}</SettingName>
        <select
          name="language"
          value={settings.language}
          onChange={e => {
            setSettingsItem('language', e.target.value);
          }}
        >
          <option value="EN">English</option>
          <option value="PT">PortuguÃªs</option>
        </select>
      </label>
    </SettingsItem>

    <SettingsItem>
      <ResetBtn onClick={() => resetSettings()}>
      {l(`Reset to default settings`)}
      </ResetBtn>
    </SettingsItem>
  </Container>
);

const Container = styled.div`
  padding: 20px 10px;
  width: 100%;
`;
const SettingsItem = styled.div`
  margin-top: 20px;
`;
const SettingName = styled.p`
  margin-bottom: 10px;
  font-size: 18px;
`;

const ResetBtn = styled.button`
  margin-top: 50px;
  width: 100%;
  height: 3em;
`;
