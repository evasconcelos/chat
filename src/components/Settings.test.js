import React from 'react';
import { shallow } from 'enzyme';
import Settings from './Settings';
import toJson from 'enzyme-to-json';

const fakeSettings = {
  lightTheme: true,
  clockDisplay12h: true,
  ctrlEnterSendMsg: true,
  uid: '1234567890',
  userName: 'Guest',
  language: 'EN',
};
const wrapper = shallow(
  <Settings
    settings={fakeSettings}
    setSettingsItem={jest.fn()}
    resetSettings={jest.fn()}
  />
);
it('Matches snapshot', () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});
