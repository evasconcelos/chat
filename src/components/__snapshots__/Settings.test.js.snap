// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<styled.div>
  <h1>
    Settings
  </h1>
  <styled.div>
    <label>
      <styled.p>
        User name
      </styled.p>
      <input
        maxLength="20"
        name="userName"
        onChange={[Function]}
        type="text"
        value="Guest"
      />
    </label>
  </styled.div>
  <styled.div>
    <styled.p>
      Interface Color
    </styled.p>
    <label>
      <input
        checked={true}
        name="interfaceColor"
        onChange={[Function]}
        type="radio"
        value="light"
      />
      Light
    </label>
    <label>
      <input
        checked={false}
        name="interfaceColor"
        onChange={[Function]}
        type="radio"
        value="dark"
      />
      Dark
    </label>
  </styled.div>
  <styled.div>
    <styled.p>
      Clock Display
    </styled.p>
    <label>
      <input
        checked={true}
        name="clockDisplay12h"
        onChange={[Function]}
        type="radio"
        value="12"
      />
      12 Hours
    </label>
    <label>
      <input
        checked={false}
        name="clockDisplay12h"
        onChange={[Function]}
        type="radio"
        value="24"
      />
      24 Hours
    </label>
  </styled.div>
  <styled.div>
    <styled.p>
      Send messages on CTRL+ENTER
    </styled.p>
    <label>
      <input
        checked={true}
        name="ctrlEnterSendMsg"
        onChange={[Function]}
        type="radio"
        value="yes"
      />
      On
    </label>
    <label>
      <input
        checked={false}
        name="ctrlEnterSendMsg"
        onChange={[Function]}
        type="radio"
        value="no"
      />
      Off
    </label>
  </styled.div>
  <styled.div>
    <label>
      <styled.p>
        Language
      </styled.p>
      <select
        name="language"
        onChange={[Function]}
        value="EN"
      >
        <option
          value="EN"
        >
          English
        </option>
        <option
          value="PT"
        >
          PortuguÃªs
        </option>
      </select>
    </label>
  </styled.div>
  <styled.div>
    <styled.button
      onClick={[Function]}
    >
      Reset to default settings
    </styled.button>
  </styled.div>
</styled.div>
`;
